crates/holochain/src/core/sys_validate.rs

```diff
@@ -280,6 +267,33 @@ pub fn check_prev_type(action: &Action, prev_action: &Action) -> SysValidationRe
     }
 }
 
+/// Check that the author didn't change between actions
+pub fn check_prev_author(action: &Action, prev_action: &Action) -> SysValidationResult<()> {
+    // Agent updates will be valid when DPKI support lands
+    #[cfg(feature = "dpki")]
+    if let Action::Update(Update {
+        entry_type: EntryType::AgentPubKey,
+        entry_hash,
+        ..
+    }) = prev_action
+    {
+        // When the agent key has been updated,
+        let new_author: AgentPubKey = entry_hash.clone().into();
+        if *action.author() == new_author {
+            return Ok(());
+        }
```

Comment:
I feel like this is backwards, since the only equality check is for the NEW author.

Shouldn't we be checking that the update was signed by the OLD author? Otherwise anybody could sign in a new key.

Reply:
I think this is right: this is the case when the previous action was an Update, and the current action is something else. So the Update was already checked in a previous validation run, in the same way as any other action. This case is the only time that a discontinuity in authorship is allowed, since the Update and everything before it had to be signed by the old author, but everything after this point needs to be signed by the new key specified in the Update.

However now that you mention it this makes me realize that this is wrong in another way, because this would still allow the next action after an update to be by the old OR the new author, so it needs to invalidate if it's by the OLD author.

Reply:
@maackle - awesome, cool, yeah I missed that part, thinking this current action was the update. I'll change my review to an approve :+1:

Reply:
@neonphog wanna double check the change I made here?

Reply:
@maackle - looks technically correct, if a little hard to mentally parse : )